server/server.go

```diff
@@ -69,7 +69,8 @@ func initializeHandlers() {
 	}
 }
 
-func applyDecoratorsAndHandle(p string, h routes.IntermediateHandler, ds []routes.Decorator) {
-	h = routes.ApplyDecorators(h, ds...)
+func applyDecoratorsAndHandle(p string, ih routes.IntermediateHandler, ds []routes.Decorator) {
+	ih = routes.ApplyDecorators(ih, ds...)
+	h := http.StripPrefix(p, ih)
```

Comment:
Please do that using a decorator, then use that decorator in `server.go` where there is a list of decorators applied to all routes.